src/web/api/src/test/java/org/locationtech/geogig/spring/controller/SendObjectControllerTest.java

```diff
@@ -0,0 +1,128 @@
+        repoProvider.getTestRepository("repo1").initializeRpository();
+        MockHttpServletRequestBuilder post =
+                MockMvcRequestBuilders.post("/repos/repo1/repo/sendobject");
+        perform(post).andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.TEXT_PLAIN))
+                .andExpect(content().string(""));
+        repo.close();
+    }
+
+    @Test
+    public void testInvalidObjectData() throws Exception {
+        Repository repo = repoProvider.createGeogig("repo1", null);
+        repoProvider.getTestRepository("repo1").initializeRpository();
+        String garbage = "This is just a bunch of random data that should cause the Object parser to choke";
+        // payload needs to end with byte value 0 for the serilaizer to know the data is finished
+        byte[] content = new byte[garbage.length() + 1];
+        System.arraycopy(garbage.getBytes(), 0, content, 0, content.length-1);
+        content[content.length-1] = 0;
+        MockHttpServletRequestBuilder post =
+                MockMvcRequestBuilders.post("/repos/repo1/repo/sendobject").content(content);
+        perform(post).andExpect(status().isBadRequest())
+                .andExpect(content().contentType(MediaType.APPLICATION_XML))
+                .andExpect(content().string(containsString("Unrecognized object header:")));
+        repo.close();
+    }
+
+    @Test
+    public void testSendObjects() throws Exception {
+        Repository repo = repoProvider.createGeogig("repo1", null);
+        repoProvider.getTestRepository("repo1").initializeRpository();
+        new TestData(repo).init("testGeoGig", "geogig@geogig.org").loadDefaultData();
+        Iterator<RevCommit> call = repo.command(LogOp.class).call();
+        assertTrue(call.hasNext());
+        // get the Object serializer
+        final ObjectSerializingFactory serialFac = DataStreamSerializationFactoryV1.INSTANCE;
+        while (call.hasNext()) {
+            RevCommit next = call.next();
+            // serialize the RevCommit
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            // need to write the 20 byte ObjectId first?
+            byte[] oidHeader = next.getId().getRawValue();
+            assertEquals(20, oidHeader.length);
+            baos.write(oidHeader);
+            serialFac.write(next, baos);
+            // build the API request
+            MockHttpServletRequestBuilder post =
+                    MockMvcRequestBuilders.post(
+                            "/repos/repo1/repo/sendobject").content(baos.toByteArray());
+            perform(post).andExpect(status().isOk())
+                    .andExpect(content().contentType(MediaType.TEXT_PLAIN))
+                    .andExpect(content().string(""));
+            // TODO: The only way to know that this worked is to inspect the INFO logs as the API
+            // response is just a 200 OK wether or not objects were ingested.
```

Comment:
This can probably be tested by creating a few RevObjects that don't exist inside the repository, then see if they are inside the object database after the request is completed.